src/ProcessTools.cpp

```diff
@@ -280,6 +280,46 @@ namespace {
         int ChildPID = 0;
     };//ProcessInfo
 
+    /// @brief An internal only methods for tokenizing command line args and respecting quotes.
+    /// @arg Arguments A string of text to tokenize.
+    /// @return A vector of tokens splits on spaces and tabs, but not breaking items single or double quoted.
+    [[nodiscard]]
+    std::vector<Mezzanine::String> TokenizeProcessArguments(const StringView Arguments)
+    {
+        std::vector<Mezzanine::String> ArgVector;
+
+        String TempStr;
+        for( StringView::iterator StrIt = Arguments.begin() ; StrIt < Arguments.end() ; ++StrIt )
+        {
+            auto ConsumeQuote = [&](const Char8 Quote) {
+                if( (*StrIt) == Quote ) {
+                    ++StrIt;
+                    while( StrIt != Arguments.end() && (*StrIt) != Quote && (*(StrIt-1)) != '\\' ) {
+                        TempStr.push_back(*StrIt);
+                        ++StrIt;
+                    }
+                    ++StrIt;
```

Comment:
If the failing condition is that StrIt is at the end, then couldn't there be problems if you increment it further?

Reply:
Yes.